### PR TITLE
Change dotnet run to call the muxer

### LINKED ISSUE #1606: `dotnet run` needs to pass NuGet cache probing path to dotnet.exe
@schellap, this is tracking the work to provide dotnet.exe with an additional probing path in the `dotnet run` scenario. Can you provide a link for the relevant dotnet.exe issue so we can take a dependency on it?

/cc @gkhanna79

### PR DESCRIPTION
**Pull request by brthor**
Fixes #1606 

this will pass --additionalprobingpath to the host when using dotnet run.
@schellap any way you could test this in your branch?

cc @gkhanna79

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/1851)

### CODE REVIEW

File: src/dotnet/commands/dotnet-run/RunCommand.cs [hunk 1]
```diff
@@ -124,6 +124,12 @@ private int RunExecutable()
                 return result;
             }
 
+            // Add Nuget Packages Probing Path
+            var nugetPackagesRoot = _context.PackagesDirectory;
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-run/RunCommand.cs]: Can you create a method AddNuGetPackagesProbingPath and encapsulate these four lines of code and call it here. Then remove the comment. Much cleaner.

Also, no tests?

**Comment by livarcocc** [comment]
Squash please.

File: src/dotnet/commands/dotnet-run/RunCommand.cs (hunk 1 above)
Comment (brthor) [inline src/dotnet/commands/dotnet-run/RunCommand.cs]: Intention here was to give something to quickly test the scenario.

As this fleshes out in the host I'll make better investments here.

Comment (gkhanna79) [inline src/dotnet/commands/dotnet-run/RunCommand.cs]: To confirm, other commands like dotnet-compile or dotnet-build, etc, will be .NETCore apps and will be running against the SharedFramework when activated by dotnet.dll. Correct @brthor @piotrpMSFT ?

Comment (gkhanna79) [inline src/dotnet/commands/dotnet-run/RunCommand.cs]: If an application is under development and not published, folks will still attempt to execute it via "dotnet <app.dll>". How is the hint being passed to the muxer to actually map that to "dotnet run <app.dll>" and invoke CLI @anurse @piotrpMSFT @brthor @schellap ?

Comment (brthor) [inline src/dotnet/commands/dotnet-run/RunCommand.cs]: I believe the work to make cli a SharedFramework app is underway. 

When doing "dotnet x.dll" it seems like we have two separate scenarios

Scenario 1 maps this to dotnet exec x.dll and shouldn't depend on the nuget cache
Scenario 2 maps this to dotnet run x.dll and will depend on the nuget cache probing path

Is this a correct understanding?

**Comment by brthor** [comment]
@schellap Is additionalprobingpath before or after the dll path?

Also should this be calling exec?

**Comment by schellap** [comment]
If using the muxer the xunit support comes from `dotnet exec` with --depsfile and --additionalprobingpath before the dll.

**Comment by schellap** [comment]
@gkhanna79 do we really need the `exec` command? What does it give us? except making the code bloat a little.

Can't we simply use:

`dotnet.exe --depsfile g:\foo.deps.json --additionalprobingpath c:\ h:\bar.dll`

**Comment by schellap** [comment]
@piotrpMSFT about the above `exec` question...

**Comment by schellap** [comment]
May be to keep the common case of `dotnet.exe foo.dll` faster?

**Comment by gkhanna79** [comment]
> do we really need the exec command?

It is a straightforward way for the muxer/host to determine whether it needs to switch to CLI invocation or not. Otherwise, interpreting, using the passed arguments, to determine whether to invoke the app or  invoke CLI would be difficult.

**Comment by schellap** [comment]
Makes sense, as --version is one example. @brthor you must use `exec`

**Comment by brthor** [comment]
updated to use "exec"

Not working on portable apps with native components. Filed an issue need to investigate.

cc @schellap

**Comment by schellap** [comment]
> Not working on portable apps with native components. Filed an issue need to investigate.

Just to be clear, for the xunit case, you mean the `hostpolicy.dll` itself needs to be resolved out of the NuGet cache?

**Comment by brthor** [comment]
@schellap Yes that is the case for xunit, because neither the test project nor test runner carry a host. 

That is not the same issue we are seeing with PortableAppWithNative though.

**Comment by brthor** [comment]
@dotnet-bot test this please

**Comment by brthor** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please
@dotnet-bot test Windows_2016 x64 Release Build please
@dotnet-bot test  Windows_2016 x86 Debug Build please
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by brthor** [comment]
Fixes #1905

**Pull request merged**